### PR TITLE
Attempt to check and use configured juju-ha-space for mongo space bef…

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

If a controller configuration has a value set for _juju-ha-space_, we want to use this value for the Mongo space in controller info instead of attempting to automatically determine such a space.

## QA steps

Accompanying unit test changes.

## Documentation changes

No.

## Bug reference

N/A.

### CODE REVIEW

**Comment by manadart** [comment]
[Filter State Addresses by HA Space](https://trello.com/c/CJzf5lmC/44-filter-state-addresses-by-ha-space)

File: worker/peergrouper/worker.go [hunk 1]
```diff
@@ -435,53 +437,86 @@ func machineAddresses(machines map[string]*machineTracker) [][]network.Address {
...
 	}
 
-	return unset, nil
+	return unsetSpace, nil
+}
+
+// trySetMongoSpaceFromConfig checks whether a value was supplied for
+// juju-ha-space in the controllers config.
+// If so, an attempt is made to use that value as the Mongo space name.
+func (w *pgWorker) tryGetMongoSpaceFromConfig(addrs [][]network.Address) (network.SpaceName, error) {
```
Comment (jameinel) [inline worker/peergrouper/worker.go]: why are you passing in the addresses here? Aren't you just getting the space name?

File: worker/peergrouper/worker.go [hunk 2]
```diff
@@ -357,8 +359,8 @@ func inStrings(t string, ss []string) bool {
 	return false
 }
 
-// apiserverHostPorts returns the host-ports for each apiserver machine.
-func (w *pgWorker) apiserverHostPorts() map[string][]network.HostPort {
+// apiServerHostPorts returns the host-ports for each apiserver machine.
+func (w *pgWorker) apiServerHostPorts() map[string][]network.HostPort {
```
[the remark this answers is not in the document]
Comment (manadart) [reply worker/peergrouper/worker.go]: Yes; non-functional. Pedantry to please the spell checker.

File: worker/peergrouper/worker.go (hunk 1 above)
Comment (manadart) [reply worker/peergrouper/worker.go]: Ah, yes. I added then removed code in that method, but left the argument; will fix.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**